src/families/tron/bridge/js.js

```diff
@@ -259,20 +240,90 @@ const currencyBridge: CurrencyBridge = {
   scanAccountsOnDevice
 };
 
-const createTransaction = a => {
-  throw new CurrencyNotSupported("tron currency not supported", {
-    currencyName: a.currency.name
-  });
-};
+const createTransaction = () => ({
+  family: "tron",
+  amount: BigNumber(0),
+  mode: "send",
+  duration: 3,
+  recipient: "",
+  networkInfo: null
+});
 
 const updateTransaction = (t, patch) => ({ ...t, ...patch });
 
-const getTransactionStatus = a =>
-  Promise.reject(
-    new CurrencyNotSupported("tron currency not supported", {
+const getEstimatedFees = async t => {
+  // see : https://developers.tron.network/docs/bandwith#section-bandwidth-points-consumption
+  // 1. cost around 200 Bandwidth, if not enough check Free Bandwidth
+  // 2. If not enough, will cost some TRX
+  // 3. normal transfert cost around 0.002 TRX
+  // Special case: If activated an account, cost around 0.1 TRX
+  let fees = BigNumber(0);
+
+  // Calculate Bandwidth :
+  if (t.networkInfo) {
+    const {
+      freeNetUsed = 0,
+      freeNetLimit = 0,
+      NetUsed = 0,
+      NetLimit = 0
+    } = t.networkInfo;
+    const bandwidth = freeNetLimit - freeNetUsed + NetLimit - NetUsed;
+
+    if (bandwidth < AVERAGE_BANDWIDTH_COST) {
+      fees = fees.plus(BigNumber(2000));
+      // Cost is around 0.002 TRX
+    }
+  }
+
+  // Active the account if there's no data, that require fees
+  if (t.recipient) {
+    const recipientAccount = await fetchTronAccount(t.recipient);
+    if (recipientAccount.length === 0) {
+      fees = fees.plus(BigNumber(100000));
+      // Cost is around 0.1 TRX
+    }
+  }
+
+  return fees;
+};
+
+const getTransactionStatus = async (a, t) => {
+  const errors = {};
+  const warnings = {};
+  const tokenAccount = !t.subAccountId
+    ? null
+    : a.subAccounts && a.subAccounts.find(ta => ta.id === t.subAccountId);
+  const account = tokenAccount || a;
+
+  if (!t.recipient) {
+    errors.recipient = new RecipientRequired("");
+  } else if (!(await validateAddress(t.recipient))) {
+    errors.recipient = new InvalidAddress("", {
       currencyName: a.currency.name
-    })
-  );
+    });
+  }
+
+  const estimatedFees = errors.recipient
+    ? BigNumber(0)
+    : await getEstimatedFees(t); //TBD
+  const amount = BigNumber(t.amount || 0);
```

Comment:
why are you fallback to `0` ?
`amount` type is required:
```
export type TransactionCommon = {|
  amount: BigNumber,
```

Reply:
probably legacy code where amount used to be ?BigNumber, we can definitely get rid of `|| 0`